### DISCUSSION TITLE
Obtaining a MutableStmtGraph from Source

### DISCUSSION THREAD

**Discussion by lbettels** [soot-oss/SootUp#774] [Q&A] [answered]
Is it possible to obtain a `MutableStmtGraph` from something like a `MethodBody`?
In my project I need to explicitly set two additional nodes in the graph and `Body.getStmtGraph()` unfortunately only returns an immutable Graph.
It would also be fine for me to built a `MutableStmtGraph` from an i.e. `ForwardingStmtGraph` if someone could point me in the right direction of how to do it.
Cheers

**Comment by swissiety** [comment, marked as answer]
Hi @lbettels,
It is - just pass the StmtGraph you get from the Body into the constructor
 `new MutableStmtGraph( existing_possibly_immutable_StmtGraph );`

depending on your use case you could implement and configure a `BodyInterceptor` as well - this would hook into the process before returning a Body via View - then no additional rebuilding of the MutableStmtGraph is required as it could reuse the MutableStmtGraph instance that is passed via the BodyBuilder

**Reply by lbettels** [reply]
Hey thank you so much I just oversaw it in the documentation.
Would you be willing to also quickly help me with some guidance to inverse the `StmtGraph`? I fell like I have seen a way to achieve this but I cant find it anymore in the documentation.

Cheers
